.github/workflows/union_tests.yml

```
@@ -0,0 +1,233 @@
+        uses: actions/cache@v4
+        with:
+          path: |
+            ${{ env.ROOT }}/rust-bitvmx-client/target
+            ${{ env.ROOT }}/rust-bitvmx-storage-backend/target
+            ${{ env.ROOT }}/rust-bitvmx-bitcoin-rpc/target
+            ${{ env.ROOT }}/rust-bitvmx-settings/target
+            ${{ env.ROOT }}/BitVMX-CPU/target
+            ${{ env.ROOT }}/rust-bitvmx-broker/target
+            ${{ env.ROOT }}/rust-bitcoind/target
+            ${{ env.ROOT }}/rust-bitvmx-key-manager/target
+            ${{ env.ROOT }}/rust-bitvmx-protocol-builder/target
+            ${{ env.ROOT }}/rust-bitcoin-indexer/target
+            ${{ env.ROOT }}/rust-bitvmx-transaction-monitor/target
+            ${{ env.ROOT }}/rust-bitvmx-operator-comms/target
+            ${{ env.ROOT }}/rust-bitvmx-wallet/target
+            ${{ env.ROOT }}/rust-bitcoin-coordinator/target
+            ${{ env.ROOT }}/rust-bitvmx-job-dispatcher/target
+          key: ${{ runner.os }}-rust-bitvmx-client-target-${{ hashFiles('rust-bitvmx-client/Cargo.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-rust-bitvmx-client-target-
+
+      - name: Build project
+        run: | 
+          cd ${ROOT}/rust-bitvmx-client
+          cargo build --release
+          cargo build --release --example union
+          
+      - name: Run integration examples
+        run: |
+          set -e
+
+          run_example() {
+            echo "::group::🧪 Running example: $1"
+
+            # Set up log directory
+            LOG_DIR="./logs/$1"
+            mkdir -p "$LOG_DIR"
+
+            # Start bitcoind using cargo
+            echo "Starting bitcoind with setup_bitcoin_node..."
+            RUST_BACKTRACE=1 cargo run --release --example union setup_bitcoin_node > $LOG_DIR/bitcoind.log 2>&1 &
+            BITCOIND_PID=$!
+            echo "Bitcoind started with PID: $BITCOIND_PID"
+
+            # Wait for bitcoind to be ready (max 30 seconds)
+            echo "Waiting for bitcoind to be ready..."
+            BITCOIND_READY=false
+            for i in {1..30}; do
+              if curl --silent --user foo:rpcpassword \
+                --data-binary '{"jsonrpc":"1.0","id":"healthcheck","method":"getblockchaininfo","params":[]}' \
+                -H 'content-type: text/plain;' http://127.0.0.1:18443/ 2>/dev/null | grep -q "result"; then
+                echo "✅ Bitcoind is ready (attempt $i)"
+                BITCOIND_READY=true
+                break
+              fi
+              echo "⏳ Waiting for bitcoind... (attempt $i/30)"
+              sleep 1
+            done
+            
+            if [ "$BITCOIND_READY" = false ]; then
+              echo "❌ Bitcoind failed to start after 30 seconds"
+              echo "=== Bitcoind logs ==="
+              cat $LOG_DIR/bitcoind.log || echo "No logs found"
+              exit 1
+            fi
+
+            # Clean previous state for this example
+            rm -rf /tmp/broker_p2p_6118* /tmp/regtest/ || true
```

```suggestion
            rm -rf /tmp/regtest/ || true
```
broker files does not exists anymore